### ISSUE TITLE
Handling files `.md` properly.

### ISSUE THREAD

**Issue by Meno-101** [remarcmij/arab-api#1]
Hello Jim, I have started slowly with the project,
I ran it and I saw some errors come from `arab-api/src/api/parser.ts` as
```
UnhandledPromiseRejectionWarning: Error: Unrecognized field name: 'source'.
```

I knew how to fix it, but I didn't know the requirement yet.

I will start submitting some PRs this week until we have our first walk through, I will be a bit busy tomorrow, and may be the day after (not sure).

Excited for the beginnings :)

**Comment by remarcmij** [comment]
@Meno-101

Hi Jawhar, I've also started picking up again the app the last view days. I'm also looking forward to working with you on this. It is probably more convenient to work on shared repos rather than using forks: easier to keep everything in sync. Therefore I have added you as a collaborator to the API and client repos (perhaps it is better to combine them in a single repo and maintain separate private repo for the content).

As I am revisiting the code, I am making some (minor) changes using a stricter form of TypeScript type checking. I'm also keen on using TypeScript 3.7 once it becomes available and is supported in VSCode. Especially Optional Chaining is nice. https://devblogs.microsoft.com/typescript/announcing-typescript-3-7-beta/

What is also badly needed is some documentation. I will start with describing the backend and the (markdown) content format. I hope that I have something that I can share tonight or otherwise tomorrow.
